pkg/openstack/loadbalancer_sg.go

```diff
@@ -63,7 +63,7 @@ func applyNodeSecurityGroupIDForLB(network *gophercloud.ServiceClient, svcConf *
 			continue
 		}
 
-		listOpts := neutronports.ListOpts{DeviceID: serverID}
+		listOpts := neutronports.ListOpts{}
```

Comment:
IMO this is an expensive request. I wonder if we can do better. The reason why I did it this way is to limit the number of API calls, but now this list might be really big in some deployments.

An alternative could be to make two queries: one that lists by SG and one that list by server ID.

Reply:
This will definitely increase the load

Reply:
an alternative could be another loop where we filter by SGs and update the ports... but it's also some load.